pkg/webui/console/store/middleware/logics/device-repository.js

```diff
@@ -104,15 +106,18 @@ const getRepositoryPayloadFormattersLogic = createRequestLogic({
         appId,
         version,
       )
-      const repositoryPayloadFormatters = {
+      repositoryPayloadFormatters = {
         ...uplinkDecoder,
         ...downlinkDecoder,
         ...downlinkEncoder,
       }
       return repositoryPayloadFormatters
     } catch (error) {
-      // There is no codec for this end device.
-      return null
+      if (isNotFoundError(error) && typeof repositoryPayloadFormatters !== 'undefined') {
+        return repositoryPayloadFormatters
+      }
+
+      throw error
```

Comment:
Still not happy with this logic:

1. It runs three requests successively that could be run in parallel
2. If the first request errors out with a 404, the other requests will also not run (a thrown exception will stop further execution of the current `try`-block)

The if condition here can never be true because the error would occur before the assignment of `repositoryPayloadFormatters`

You will likely want something like this:
```js
  const ignoreNotFound = async (action, ...params) => {
    try {
      return await action(...params)
    } catch (e) {
      if (!isNotFound(e)) {
        throw e
      }
    }

    return undefined
  }

  const results = await Promise.all([
    ignoreNotFound(tts.Applications.Devices.Repository.getUplinkDecoder, ...),
    ignoreNotFound(tts.Applications.Devices.Repository.getDownlinkDecoder, ...),
    ignoreNotFound(tts.Applications.Devices.Repository.getDownlinkEncoder, ...),
  ])
  
  return {
    ...results[0],
    ...results[1],
    ...results[2],
  }
```